sweagent/environment/swe_env.py

```diff
@@ -135,6 +137,14 @@ def __init__(self, args: EnvironmentArguments):
         self.image_name = args.image_name
         self._reset_container()
 
+        # Prepare image tag prefix for cached task environments
+        if self.args.cache_task_images:
+            logger.info("Task environment caching enabled")
+            tag = f"{self.args.data_path.replace('/', '_')}__{self.args.split}__{self.args.base_commit or 'head'}__"
```

Comment:
Data path can now be all kinds of things, including the full text of the problem statement. I could push some code to work around these things.

Though I also wonder if `data_path` is really what we should make this depend on. Perhaps we could rather intercept the actual setup stages and hash the setup config or something like that?

Reply:
Some dataset fingerprints would be even better, I agree. Also, there is a limit of 128 characters for the docker image tag, so this would help stay within the limit with any dataset. I will try to implement the idea with the config hash.